### ISSUE TITLE
Question: Is there a way to specify our own icon?

### ISSUE THREAD

**Issue by MattL-NZ** [aryzhov/flutter-expandable#1]
I would like to specify my own icon and also the direction in which the icon faces.

![image](https://user-images.githubusercontent.com/2846782/52824949-a85e2480-311f-11e9-91df-3c6073ac43cb.png)
![image](https://user-images.githubusercontent.com/2846782/52824921-8c5a8300-311f-11e9-8059-b2b333bc2714.png)

It would also be great to have the ability to change the color of the Expandable Panels as well as covering the icon in that color.

**Comment by aryzhov** [comment]
The way to do it is to use a custom widget instead of `ExpandableIcon`.
The demo has a button that changes its text based on the expanded/collapsed state:
```
                Builder(
                  builder: (context) {
                    var exp = ExpandableController.of(context);
                    return MaterialButton(
                      child: Text(exp.expanded ? "COLLAPSE": "EXPAND",
                        style: Theme.of(context).textTheme.button.copyWith(
                          color: Colors.deepPurple
                        ),
                      ),
                      onPressed: () {
                        exp.toggle();
                      },
                    );
                  },
                ),
```
You can use an icon instead of text, of course. If you want a nice rotating animation though, that's not provided out-of-the box. That would be a valid feature request.

**Comment by aryzhov** [comment]
The ability to specify a custom icon has been implemented in version 4.1.1.
